### ISSUE TITLE
/run shows host system

### ISSUE THREAD

**Issue by damoon** [weaveworks/weave#1232]
without weave `docker run -ti --rm postgres:9.3 ls -1 /run` shows `exim3 lock postgresql utmp`.
after running
```
weave launch
weave launch-dns
weave launch-proxy
`weave env`
```
the same command shows the /run of the hostsystem.

i found this, because the default postgres image cant't be started with weave.
it complains:
```
chmod: cannot access ‘/run/postgresql’: No such file or directory
```
when is entrypoint tries to set the permissions.

**Comment by rade** [comment]
Reproduced the odd `ls -1 /run` output. This is really weird. Thanks for reporting this!

Running a `docker inspect` on the container resulting from `docker run -ti --rm postgres:9.3 /bin/bash` shows
````
    "Volumes": {
        "/var/lib/postgresql/data": "/var/lib/docker/volumes/20a649afd0aebbf7e8c80bed6069f18e31e55ff180c597eebda457560124df31/_data"
    },
````
but when run via the proxy we get
````
    "Volumes": {
        "/hostproc": "/proc",
        "/var/lib/postgresql/data": "/var/lib/docker/volumes/eacf8f61cb57fa7699281e6b37aa2c2513b0a946642ceee3e277aaf6e454ff4a/_data",
        "/var/run": "/var/run",
        "/w": "/var/lib/docker/volumes/d707d68299fa45691265940567d60b352bc1841ac3977ffa6f13637e7dd23720/_data"
    },
````

**Comment by rade** [comment]
Ah, I know why this happens. We are using `--volumes-from` to mount the volumes of the weaveproxy container. That drags in that container's volume mounts. Which is very bad indeed, not just because of the kind of breakages reported here but also because it gives the container access to files on the host that they should not be able to see.

There doesn't appear to be a way to *selectively* mount volumes from another container. So I think we are going to have to create a separate instance of the weavexec container, started without any volume mounts of its own.

**Comment by rade** [comment]
This got fixed by #1238. I have no idea why it didn't get closed automatically.
